lib/checkmemoryleak.cpp

```diff
@@ -551,7 +551,9 @@ void CheckMemoryLeakInFunction::checkReallocUsage()
 
                 // Check that another copy of the pointer wasn't saved earlier in the function
                 if (Token::findmatch(scope->bodyStart, "%name% = %varid% ;", tok, tok->varId()) ||
-                    Token::findmatch(scope->bodyStart, "[{};] %varid% = %name% [;=]", tok, tok->varId()))
+                    Token::findmatch(scope->bodyStart, "[{};] %varid% = %name% [;=]", tok, tok->varId()) ||
```

Comment:
I believe you can rewrite 3 expressions in 1:

     Token::findmatch(scope->bodyStart, "[{};] %varid% = *| %name% .| %name%| [;=]", tok, tok->varId())

Reply:
That picks up some other patterns as well, but they're either also okay or syntax errors — so that's fine. Done, and the test suite still passes.